libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslFileUtil.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.ssl;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.AccessDeniedException;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.security.AccessControlException;
+import java.security.GeneralSecurityException;
+import java.security.UnrecoverableKeyException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Utility methods for common file handling in SSL configuration
+ */
+final class SslFileUtil {
```

Comment:
This is a new class, mostly to handle generating exceptions for particular file related problems.